fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckService.kt

```diff
@@ -110,8 +118,23 @@ class DefaultReviewQualityCheckService(
     override fun selectedTabUrl(): String? =
         browserStore.state.selectedTab?.content?.url
 
-    private inline fun <reified T : Enum<T>> String.asEnumOrDefault(defaultValue: T? = null): T? =
-        enumValues<T>().firstOrNull { it.name.equals(this, ignoreCase = true) } ?: defaultValue
+    override suspend fun productRecommendations(): ProductRecommendation? =
+        withContext(Dispatchers.Main) {
+            suspendCoroutine { continuation ->
+                browserStore.state.selectedTab?.let { tab ->
+                    tab.engineState.engineSession?.requestProductRecommendations(
+                        url = tab.content.url,
+                        onResult = {
+                            continuation.resume(it.firstOrNull())
```

Comment:
let's include a comment why we only care about the first recommendation and ignoring whatever else the API returns